pkg/lib/oauth/handler/handler_token.go

```diff
@@ -130,6 +131,10 @@ type TokenHandlerOfflineGrantService interface {
 	GetOfflineGrant(ctx context.Context, id string) (*oauth.OfflineGrant, error)
 }
 
+type TokenHandlerRateLimiter interface {
```

Comment:
Where did we apply the Per-IP rate limit?

Reply:
I think this comment is related to the below comment? https://github.com/authgear/authgear-server/pull/5317#discussion_r2261856640

Will check it before opening db connection.